geotiepoints/simple_modis_interpolator.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# Copyright (c) 2021 Python-geotiepoints developers
+#
+# This file is part of python-geotiepoints.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Interpolate MODIS 1km navigation arrays to 250m and 500m resolutions.
+
+The code used here is a rewrite of the IDL function ``MODIS_GEO_INTERP_250``
+used by Liam Gumley. It has been modified to convert coordinates to cartesian
+(X, Y, Z) coordinates first to avoid problems with the anti-meridian and poles.
+This code was originally part of the CSPP Polar2Grid project, but has been
+moved here for integration with Satpy and newer versions of Polar2Grid.
+
+This algorithm differs from the one in ``modisinterpolator`` as it only
+requires the original longitude and latitude arrays. This is useful in the
+case of reading the 250m or 500m MODIS L1b files or any MODIS L2 files without
+including the MOD03 geolocation file as there is no SensorZenith angle dataset
+in these files.
+
+"""
+
+from functools import wraps
+
+import numpy as np
+from scipy.ndimage.interpolation import map_coordinates
+from .geointerpolator import lonlat2xyz, xyz2lonlat
+
+try:
+    import dask.array as da
+except ImportError:
+    # if dask can't be imported then we aren't going to be given dask arrays
+    da = None
+
+try:
+    import xarray as xr
+except ImportError:
+    xr = None
+
+# MODIS has 10 rows of data in the array for every scan line
+ROWS_PER_SCAN = 10
+
+
+def scanline_mapblocks(func):
+    """Convert dask array inputs to appropriate map_blocks calls.
+
+    This function, applied as a decorator, will call the wrapped function
+    using dask's ``map_blocks``. It will rechunk inputs when necessary to make
+    sure that the input chunks are entire scanlines to avoid incorrect
```

Comment:
Should it also make sure an integer number of scans is used?

Reply:
Done.